ls1intum/Artemis
# Disable save button in code editor and replace with auto save

### Checklist

### Motivation and Context


Some students had problems during the exam to differentiate the "save" and "submit" button of programming exercises, which lead to unsubmitted submissions in these exams.

### Description

This PR solves the issue by replacing the save button with an autosave functionality. The student still has to submit the code in order to get it assessed, but the confusion between these buttons should be solved since the submitting is now the only option.

### Steps for Testing

To test this PR, you need: 

- 1x **Exam** with 1x **Exercise Group** with 1x **Programming Exercise** with **Online Code Editor enabled**
- 1x **Regular Programming Exercise** with **Online Code Editor enabled**

1. With the student account, participate in the regular programming exercise **and** the exam exercise using the **Online Code Editor**
2. Change some lines in the files of the exercise.
3. Check that the files are saved every 30 seconds as well as when leaving the site:
 - Closing the browser tab/ reloading the page should show a warning, if there are unsaved changes
 - Navigating somewhere in Artemis (changing exercise, going to the main page) should save the current state
4. Additionally make sure that the UI intuitively makes clear that the student has to submit his changes in order to get them assessed.

### Test Coverage



code-editor-actions.component.ts: 78% -> 81%


### Screenshots


![Bildschirmfoto 2021-06-25 um 01 00 18](https://user-images.githubusercontent.com/38322605/123342807-bc688b00-d550-11eb-86e3-a65a353462c7.png)
